interpreter/cling/lib/MetaProcessor/CMakeLists.txt

```diff
@@ -10,7 +10,16 @@ set( LLVM_LINK_COMPONENTS
   core
   support
   binaryformat
-)
+  )
+
+if (LLVM_LIB_IS_SHARED)
```

Comment:
I do not understand why we need these conditionals here? The dependencies should not depend on the build mode.

Reply:
It's been a while since I authored this change, but I guess when building Clang with `-DCLANG_LINK_CLANG_DYLIB=ON`, a single monolith shared object containing everything is made available via `libclang-cpp.so`, the same using `-DLLVM_BUILD_LLVM_DYLIB=ON` for LLVM causes a monolith `LLVM.so` containing everything to be produced.

This means instead of listing the individual library components, it suffices to link simply to libclang-cpp.so or libLLVM.so to have access to any needed symbol.